src/LayerEditor/ui/data/le_data.py

```diff
@@ -0,0 +1,296 @@
+"""Analyzis Editor static parameters
+
+.. codeauthor:: Pavel Richter <pavel.richter@tul.cz>
+"""
+
+import os
+
+from PyQt5.QtCore import QObject
+
+from LayerEditor.ui.data.block_model import BlockModel
+from LayerEditor.data.geometry_model import LayerGeometryModel
+from LayerEditor.ui.data.regions_model import RegionsModel
+from LayerEditor.ui.diagram_editor.diagram_view import DiagramView
+
+
+class LEData(QObject):
+    """Main data class for Layer Editor"""
+
+    # diagrams = []
+    # """List of diagram data"""
+    # history = None
+    # """History for current geometry data"""
+    # blocks = []
+    # """Blocks structure"""
+    # regions = None
+    # """Structure for managing regions"""
+    # curr_diagram =  None
+    # """Current diagram data"""
+    # le_serializer = None
+    # """Data from geometry file"""
+    # curr_file = None
+    # """Name of open file"""
+    # curr_file_timestamp = None
+    # """
+    # Timestamp of opened file, if editor text is
+    # imported or new timestamp is None
+    # """
+    # #path = None
+    # """Current geometry data file path"""
+    # geomop_root = os.path.dirname(os.path.dirname(
+    #               os.path.dirname(os.path.realpath(__file__))))
+    # """Path to the root directory of the GeoMop installation."""
+    # layer_heads = None
+    # # Data model for Regions panel.
+
+    def __init__(self, in_file=None):
+        super(LEData, self).__init__()
+    #     # self.history = GlobalHistory(cls)
+    #     # cls.layers = le_data.Layers(cls.history)
+    #     # cls.reinit()
+    #     self.regions = Regions()
+        self.curr_file = in_file
+        """Current file (culd be moved to config?)."""
+        self.diagram_view = DiagramView()
+        """View is common for all layers and blocks."""
+        self.regions = RegionsModel()
+        """Manages regions."""
+        if self.curr_file is None:
+            self.curr_file_timestamp = None
+        else:
+            try:
+                self.curr_file_timestamp = os.path.getmtime(in_file)
+            except OSError:
+                self.curr_file_timestamp = None
+
+        """Timestamp is used for detecting file changes while file is loaded in LE."""
+
+        geo_model = LayerGeometryModel(in_file)
```

Comment:
If LayerGeometryModel corresponds to the single layer editor's file it should possibly contain all information from the file, including DiagramView and Regions.

Reply:
Put it otherwise: what is the difference between LayerGeometryModel and LEData?